vassal-app/src/main/java/VASSAL/build/GameModule.java

```diff
@@ -2304,6 +2304,32 @@ else if (DRAWING_MOUSEOVER_INDEX_PROPERTY.equals(key)) {
     return s == null ? null : s.getPropertyValue();
   }
 
+  /**
+   * Refresh the visible portions of all currently showing maps.
+   * This is called in when the following are completed:
+   *  - Any right-click menu option on a piece
+   *  - Any Toolbar button action
+   *  - Any Toolbar Menu item selection
+   *  - Each Step replay from a log file
+   *  - Each Undo
+   *  - Each time a counter or stack is drag'n'dropped
+   *  - Each message received during on-line play
+   *  - Any click on an Action Button
+   *
+   *  These eight actions can cause changes that result in text displaying Calculated properties and Beanshell expressions
+   *  to change in places unrelated to where the action occurred, leaving these counters out of sync until a later click
+   *  in their general area.
+   */
+  public final void refreshVisibleMaps() {
+    for (final Map map : Map.getMapList()) {
+      final JComponent view = map.getView();
+      if (view.isShowing()) {
+        final Rectangle v = view.getVisibleRect();
+        view.repaint(v.x, v.y, v.width, v.height);
```

Comment:
I'm not sure it makes a difference to specify the visible rectangle, as the nullary version of `repaint()` ought to only ever repaint the visible portion anyway. Have you tried calling `view.repaint()` instead and seeing if it makes a noticeable difference?